matrix/inverse_of_matrix.py

```diff
@@ -25,24 +26,137 @@ def inverse_of_matrix(matrix: list[list[float]]) -> list[list[float]]:
     [[0.16666666666666666, -0.0625], [-0.3333333333333333, 0.25]]
     >>> inverse_of_matrix([[10, 5], [3, 2.5]])
     [[0.25, -0.5], [-0.3, 1.0]]
+
+    Demo for 3x3
+    >>> inverse_of_matrix([[2, 5, 7], [2, 0, 1], [1, 2, 3]])
+    [[2.0, 1.0, -5.0], [5.0, 1.0, -12.0], [-4.0, -1.0, 10.0]]
+    >>> inverse_of_matrix([[1, 2, 2], [1, 2, 2], [3, 2, -1]])
+    Traceback (most recent call last):
+    ...
+    ValueError: This matrix has no inverse.
+
+    More examples:
+
+    >>> inverse_of_matrix([])
+    Traceback (most recent call last):
+    ...
+    ValueError: Please provide a matrix of size 2x2 or 3x3.
+
+    >>> inverse_of_matrix([[],[]])
+    Traceback (most recent call last):
+    ...
+    ValueError: Please provide a matrix of size 2x2 or 3x3.
+
+    >>> inverse_of_matrix([[1, 2], [3, 4], [5, 6]])
+    Traceback (most recent call last):
+    ...
+    ValueError: Please provide a matrix of size 2x2 or 3x3.
+
+    >>> inverse_of_matrix([[1, 2, 1], [0,3, 4]])
+    Traceback (most recent call last):
+    ...
+    ValueError: Please provide a matrix of size 2x2 or 3x3.
+
+    >>> inverse_of_matrix([[1, 2, 3], [7, 8, 9], [7, 8, 9]])
+    Traceback (most recent call last):
+    ...
+    ValueError: This matrix has no inverse.

```

Comment:
We should test that all lines between `Traceback` and the Error are indented at least four spaces.
```suggestion
        ...
    ValueError: This matrix has no inverse.

    >>> inverse_of_matrix([])
    Traceback (most recent call last):
        ...
    ValueError: Please provide a matrix of size 2x2 or 3x3.

    >>> inverse_of_matrix([[],[]])
    Traceback (most recent call last):
        ...
    ValueError: Please provide a matrix of size 2x2 or 3x3.

    >>> inverse_of_matrix([[1, 2], [3, 4], [5, 6]])
    Traceback (most recent call last):
        ...
    ValueError: Please provide a matrix of size 2x2 or 3x3.

    >>> inverse_of_matrix([[1, 2, 1], [0,3, 4]])
    Traceback (most recent call last):
        ...
    ValueError: Please provide a matrix of size 2x2 or 3x3.

    >>> inverse_of_matrix([[1, 2, 3], [7, 8, 9], [7, 8, 9]])
    Traceback (most recent call last):
        ...
    ValueError: This matrix has no inverse.
```